hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/procedures/ShowColumnStatsOverlapProcedure.scala

```diff
@@ -142,9 +144,12 @@ class ShowColumnStatsOverlapProcedure extends BaseProcedure with ProcedureBuilde
     }
   }
 
-  def getSchema(metaClient: HoodieTableMetaClient): StructType = {
-    val schemaUtil = new TableSchemaResolver(metaClient)
-    AvroConversionUtils.convertAvroSchemaToStructType(schemaUtil.getTableAvroSchema)
+  def getAvroSchema(metaClient: HoodieTableMetaClient): Schema = {
+    new TableSchemaResolver(metaClient).getTableAvroSchema
+  }
+
+  def getStructSchema(metaClient: HoodieTableMetaClient): StructType = {
+    AvroConversionUtils.convertAvroSchemaToStructType(getAvroSchema(metaClient))
```

Comment:
This incurs `TableSchemaResolver` creation and timeline scanning again, which should be avoided.